packages/core/src/app/payment/Payment.tsx

```diff
@@ -524,15 +524,22 @@ export function mapToPaymentProps({
             isPaymentDataRequired,
         },
         errors: { getFinalizeOrderError, getSubmitOrderError },
-        statuses: { isInitializingPayment, isSubmittingOrder },
+        statuses: { isInitializingPayment, isSubmittingOrder, isStripeLinkAuthenticated },
     } = checkoutState;
 
     const checkout = getCheckout();
     const config = getConfig();
     const customer = getCustomer();
     const consignments = getConsignments();
     const { isComplete = false } = getOrder() || {};
-    const methods = getPaymentMethods() || EMPTY_ARRAY;
+    let methods = getPaymentMethods() || EMPTY_ARRAY;
+    const stripeUpePaymentMethod = methods.filter(method =>
+        method.id === 'card' && method.gateway === PaymentMethodId.StripeUPE
+    );
+
+    if (isStripeLinkAuthenticated() && !!stripeUpePaymentMethod.length){
+        methods = stripeUpePaymentMethod
+    }
```

Comment:
Why don't `isStripeLinkAuthenticated` is authenticated first? And then go down the flow of filtering cards.